### PR TITLE
Updated jaeger tracing versions

### LINKED ISSUE #3519: [Enhancement] Update Jaeger Tracing Client to v1.3.2 to support the w3c trace context
**Is your feature request related to a problem? Please describe.**

Jaeger-core supports the recent [W3C Trace Context](https://www.w3.org/TR/trace-context/) propagation format since version [1.2.0](https://github.com/jaegertracing/jaeger-client-java/blob/master/CHANGELOG.md#120-2020-03-11). W3C is the default format for many OpenTelemetry client libraries (Python, .NET, JS), supporting it would make it easier to integrate Strimzi with them.

**Describe the solution you'd like**
Strimzi currently uses [v1.1.0 of jaeger-core](https://github.com/strimzi/strimzi-kafka-operator/blob/master/pom.xml#L111). It would be great if the version could be updated to the latest v1.3.2. 

**Describe alternatives you've considered**
Other propagation formats, b3 or the jaeger default, could be used instead.

**Additional context**

### PR DESCRIPTION
**Pull request by chgl**
Signed-off-by: chgl <chgl@users.noreply.github.com>

### Type of change

- Enhancement / new feature
- Documentation

### Description

Updated Jaeger versions. Resolves #3519 

I tested the changes locally by adding `w3c` to the `JAEGER_PROPAGATION` env var of a KafkaConnect cluster which uses the updated images. The `traceparent` header was correctly inserted into the message headers.

### Checklist

- [x] Make sure all tests pass
- [x] Update documentation
- [x] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=tracing

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=tracing

**Comment by chgl** [comment]
This might be my fault for updating the jaeger service versions in the system tests without verifying it still works. There were changes in 1.16 which changed the query response, but I assumed those were backwards compatible:  https://github.com/jaegertracing/jaeger/releases/tag/v1.16.0. I can go ahead and revert the jaeger collector/query version bump which may be out of scope of  the original issue anyway.

**Comment by scholzj** [comment]
In general I do not have problem with it being bumped. But you might need to update the tests then.

@see-quick @Frawless What is your view on the Jaeger backend update? Should we keep the old version if it works? Or bump to the newer one?

**Comment by Frawless** [comment]
I think we should understand why it failed. I guess it will be small change to make tests working again with the newer version. @chgl could you please try to run the tests locally and debug it?

**Comment by chgl** [comment]
@Frawless I just ran the tests locally (Kind in WSL2) without any errors. Is it possible to access the logs of the CI run?

```console
$ mvn clean install -am -pl systemtest -DskipTests
$ mvn verify -pl systemtest -Dit.test=io.strimzi.systemtest.tracing.TracingST*
...
[WARNING] Tests run: 10, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 6,991.722 s - in io.strimzi.systemtest.tracing.TracingST
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:129] ================================================
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:130] ================================================
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:131] io.strimzi.systemtest.tracing.TracingST
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testProducerConsumerService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:04:04,723 started: 22:39:29,205 ended: 22:43:33,928
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testProducerWithStreamsService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:11:57,673 started: 22:43:33,941 ended: 22:55:31,614
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testKafkaBridgeService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:14:02,744 started: 22:55:31,618 ended: 23:09:34,362
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testProducerConsumerMirrorMakerConnectStreamsService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:09:16,311 started: 23:09:34,366 ended: 23:18:50,677
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testProducerConsumerStreamsService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:11:33,515 started: 23:18:50,688 ended: 23:30:24,203
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testProducerConsumerMirrorMakerService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:21:27,052 started: 23:30:24,211 ended: 23:51:51,263
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testConnectService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:11:31,368 started: 23:51:51,269 ended: 00:03:22,637
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testProducerConsumerMirrorMaker2Service
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:21:18,240 started: 00:03:22,641 ended: 00:24:40,881
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:133] ---------------------------------------------
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:134] testProducerService
2020-09-02 22:35:15 INFO  [TimeMeasuringSystem:136] Operation id: TEST_EXECUTION duration: 00:10:12,245 started: 00:24:40,884 ended: 00:34:53,129
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 10, Failures: 0, Errors: 0, Skipped: 1
[INFO] 
[INFO] 
[INFO] --- maven-failsafe-plugin:3.0.0-M5:verify (default) @ systemtest ---
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:analyze-only (analyze) @ systemtest ---
[INFO] No dependency problems found
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:02 h
[INFO] Finished at: 2020-09-03T00:35:19+02:00
[INFO] ------------------------------------------------------------------------
```

**Comment by Frawless** [comment]
@chgl I checked the logs and tried to reproduce it on oc cluster up and problem is in jaeger deployment.

```  Normal   Scheduled    3m24s                  default-scheduler   Successfully assigned tracing-cluster-test/my-jaeger-77f856b9f-bjwhx to localhost
  Warning  FailedMount  2m52s (x7 over 3m24s)  kubelet, localhost  MountVolume.SetUp failed for volume "my-jaeger-service-ca" : configmap references non-existent config key: service-ca.crt
  Warning  FailedMount  2m51s (x7 over 3m24s)  kubelet, localhost  MountVolume.SetUp failed for volume "my-jaeger-collector-tls-config-volume" : secrets "my-jaeger-collector-headless-tls" not found```

I guess its caused by changes done in operator and its CRD for 1.19 release. We should to same thing as we do in case of keycloak dependency - have a script to deploy it. Do you want to do it or should I open PR against your branch?

**Comment by chgl** [comment]
@Frawless thanks for checking! This might actually be related to: https://github.com/jaegertracing/jaeger-operator/issues/1029, so its OpenShift specific.

> You may want to try to set the platform to kubernetes explicitly, which should prevent the operator from attempting to provision the TLS certs.

I don't have an OpenShift cluster, but maybe this is something we can try.

**Comment by Frawless** [comment]
@chgl so it looks to me we can use 1.17 version without issues. I guess the "fix" will be part of 1.20 release so when it will be ready, we can increase jaeger version as well. Currently, we support ocp3.11 (kubernetes 1.11) so we want to have tests working on those versions.

**Comment by chgl** [comment]
@Frawless I'll downgrade the jaeger operator versions to 1.17 in this PR. Sorry for the unnecessary version bumps!

**Comment by Frawless** [comment]
@chgl no worries! I will restart tests after you change the code to see if everything is fine :)

**Comment by chgl** [comment]
New try @Frawless. Thanks!

**Comment by scholzj** [comment]
@strimzi-ci run tests profile=tracing

**Comment by scholzj** [comment]
@chgl Could you please fix the DCO sign-off? The instructions should be under the _Details_ next to the DCO status.

**Review by scholzj — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@
 * Support dynamically changeable logging configuration of Kafka Connect and Kafka Connect S2I
 * Support dynamically changeable logging configuration of Kafka brokers
 * Support dynamically changeable logging configuration of Kafka MirrorMaker2
+* Updated Jaeger tracing libraries to support the W3C trace context headers
```
Comment (ppatierno) [inline CHANGELOG.md:17]: I don't know what it is :-) ... but how much is it relevant to the user to know that the new Jaeger lib has this feature related to Kafka?

**Review by ppatierno — APPROVED** [review]
Test passed, LGTM. I left a question about changelog.

File: CHANGELOG.md (hunk 1 above)
Comment (chgl) [reply CHANGELOG.md:17]: Probably not really, there was a to-do box for updating the changelog so I added it :). But I agree that it's not really relevant to the end user. I'll remove it!

**Review by Frawless — APPROVED** [review]
LGTM, thanks for PR!

**Comment by scholzj** [comment]
Thanks for the PR.

**Comment by chgl** [comment]
Thank you and the team for your patience! And overall great job on strimzi, love using it :)

**Pull request merged**